### DISCUSSION TITLE
Fair dispatch on Rabbitmq / messenger?

### DISCUSSION THREAD

**Discussion by vbarrier** [symfony/symfony#47160] [Q&A] [answered]
Hello, 

I'm looking for a way to configure my channels in order to do Fair Dispatch with rabbitmq (https://www.rabbitmq.com/tutorials/tutorial-two-python.html) 

By default prefetch is set to 3 I would like to change to 1. There were a "prefetch_count" option in the DSN but it has been removed. 

How could I do that with messenger?

**Comment by lyrixx** [comment, marked as answer]
prefetch is useless here since we don't subscribe (`consume()`) to a channel, but we do `get()` call

**Reply by vbarrier** [reply]
messenger.yaml is also used for the workers. Do we have a way to configure prefetch for them?

**Reply by lyrixx** [reply]
the prefetch is useless here, since we use `get()` and not `consume()`

**Reply by vbarrier** [reply]
So get() act as a fair dispatch solution right?

**Reply by lyrixx** [reply]
yes, in sort of :) it only get one message per one message 👍🏼

**Reply by antoniovj1** [reply]
Hi! @vbarrier @lyrixx Do you know some way to change this behaviour?

https://github.com/symfony/symfony/discussions/50419

**Reply by ghost** [reply]
@antoniovj1 it cannot be changed because it's would require to change how symfony/amqp-messenger works (how it talks with rabbitmq). 
Actually I don't understand why do you even have problem with `prefetch_count` if you are using symfony messenger? This option was useless even before so it didn't work back then either.

**Comment by vbarrier** [comment]
Like Amazon SQS and the buffer_size params
